src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerations/AvoidMultipleEnumerations.DataFlowOperationVisitor.cs

```diff
@@ -304,19 +329,16 @@ private void UpdateGlobalValue(GlobalFlowStateDictionaryAnalysisValue value)
 
             private bool IsGetEnumeratorOfForEachLoopInvoked(IOperation operation)
             {
-                RoslynDebug.Assert(operation is ILocalReferenceOperation or IParameterReferenceOperation);
```

Comment:
Is this assert no longer true? What other operation kinds are possible here?

Reply:
It's no longer true.
In the previous revision,  it is called in this way
```
                if (!IsOperationEnumeratedByMethodInvocation(parameterOrLocalReferenceOperation, _wellKnownSymbolsInfo)
                    && !IsGetEnumeratorOfForEachLoopInvoked(parameterOrLocalReferenceOperation))
            {
             }
```
`IsOperationEnumeratedByMethodInvocation` and `IsGetEnumeratorOfForEachLoopInvoked` are accepting a local or parameter reference operation. And call `SkipLinqChainAndConversionMethod` seperately. (See the line below)
And therefore there is a duplicate call to `SkipLinqChainAndConversionMethod`. Here I extract `SkipLinqChainAndConversionMethod` to the caller to avoid this.